packages/skill-lesson/path-to-purchase/product-tiers.tsx

```diff
@@ -28,17 +28,15 @@ export const PricingTiers: React.FC<
     couponIdFromCoupon || (validCoupon ? couponFromCode?.id : undefined)
 
   const purchasedProductIds = purchases.map((purchase) => purchase.productId)
+
   return (
     <>
       {redeemableCoupon ? <RedeemDialogForCoupon /> : null}
       <div data-pricing-container="">
-        {products?.map((productWithOptions, i) => {
-          const {options, ...product} = productWithOptions
-          return (
-            <PriceCheckProvider
-              key={productWithOptions.slug}
-              purchasedProductIds={purchasedProductIds}
-            >
+        <PriceCheckProvider purchasedProductIds={purchasedProductIds}>
+          {products?.map((productWithOptions, i) => {
```

Comment:
the Price Check provider needs to be _outside_ the `products.map` so that all 3 of the `<Pricing>` components get wrapped in a single provider. Otherwise we have three separate `PriceCheckProvider` instances each of which only knows about the price it is wrapping and is not able to identifier downgrades.